ecommerce/frontend/src/components/checkout/EditCardModal.js

```diff
@@ -1,157 +1,48 @@
 import { useState } from 'react'
-import Cards from 'react-credit-cards'
 import 'react-credit-cards/es/styles-compiled.css'
-import { Modal, Form, Input } from 'antd'
+import { Modal, Form, Spin, notification } from 'antd'
+import { sleep } from '../../utils'
+import { CardModalInner } from './CardModalInner'
 
 export const EditCardModal = ({
-    cardProps = {
-        cvc: '',
-        expiry: '',
-        number: '',
-        name: '',
-    },
-    cardId = {},
-    cardName = '',
+    card,
     visible = false,
-    setVisible = () => {},
+    onCancel = () => {},
+    onSuccessfulEdit = () => {}
 }) => {
     /* This might not remounted after closing and opening again, be careful */
-    const [cvc, setCvc] = useState(cardProps.cvc)
-    const [expiry, setExpiry] = useState(cardProps.expiry)
-    const [name, setName] = useState(cardProps.name)
-    const [number, setNumber] = useState(cardProps.number)
-    const [cardNickname, setCardNickname] = useState(cardName)
-    const [focused, setFocused] = useState('')
 
     const [form] = Form.useForm()
-    const formItemLayout = {
-        labelCol: {
-            xs: {
-                span: 24,
-            },
-            sm: {
-                span: 6,
-            },
-        },
-        wrapperCol: {
-            xs: {
-                span: 24,
-            },
-            sm: {
-                span: 18,
-            },
-        },
-    }
+    const [isLoading, setIsLoading] = useState(false)
 
-    const handleInputFocus = val => setFocused(val.target.name)
+    const onEdit = async () => {
+        try {
+            setIsLoading(true)
+            await sleep(2000)
+            setIsLoading(false)
+            const fields = await form.validateFields()
+            console.log(fields)
+            onSuccessfulEdit();
+            onCancel();
+            notification.success({ message: 'You have successfully changed your credit card information!' })
+            
+        } catch (error) {
+            notification.warning({ message: 'There was an error with your request.' })
+        }
+    }
 
     return (
         <Modal
             visible={visible}
             title="Edit your credit card information"
             destroyOnClose
-            onOk={() => {
-                setVisible(false)
-            }}
-            onCancel={() => {
-                setVisible(false)
-            }}
+            onOk={onEdit}
+            onCancel={onCancel}
             cancelText="Go Back"
             okText="Edit">
```

Comment:
i think you might need to use forceRender here because of the warning from Antd in Form or Modal section I dont remember.